lib/events/events.go

```diff
@@ -319,12 +315,19 @@ func (s *bufferedSubscription) pollingLoop() {
 	}
 }
 
-func (s *bufferedSubscription) Since(id int, into []Event) []Event {
+func (s *bufferedSubscription) Since(id int, into []Event, timeout time.Duration) []Event {
 	s.mut.Lock()
 	defer s.mut.Unlock()
 
-	for id >= s.cur {
-		s.cond.Wait()
+	// Check once first before generating the TimeoutCondWaiter
+	if id >= s.cur {
+		waiter := s.cond.SetupWait(timeout)
+		for id >= s.cur {
```

Comment:
What is the reason to create `TimeoutCondWaiter` instead of setting up the timer in one call to (an adapted) `s.cond.Wait(timeout)`?